### PR TITLE
pythonPackages.whey, pythonPackages.whey-pth: init at 0.1.1

### PR DESCRIPTION
**Pull request by TyberiusPrime**
## Description of changes
Add python package 'whey'.

Whey is "a simple Python wheel builder for simple projects.".
As such it acts as a build-backend for a (small) number of packages on pypi.

Whey depends on some more python packages which are also included in this PR:
- apeye-core
- consolekit
- deprecation-alias
- dist-meta
- dom-toml
- domdf-ptyon-tools
- handy-archives
- pyproject-parser
- shippinglabel

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FRidh — APPROVED** [review]
Looks good to me. Rebase is needed, though.

**Comment by TyberiusPrime** [comment]
Thank you @FRidh, I rebased.

File: pkgs/development/python-modules/apeye-core/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [
+    hatch-requirements-txt
+  ];
+
+  dependencies = [
+    domdf-python-tools
+    idna
+  ];
+
+  nativeCheckInputs = [ ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/apeye-core/default.nix:32]: Please fix.

File: pkgs/development/python-modules/consolekit/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  dependencies = [
+    click
+    colorama
+    deprecation-alias
+    domdf-python-tools
+    mistletoe
+    typing-extensions
+  ];
+
+  nativeCheckInputs = [ ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/consolekit/default.nix:34]: ?

File: pkgs/development/python-modules/consolekit/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  meta = {
+    description = "Additional utilities for click.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/consolekit/default.nix:37]: No period

File: pkgs/development/python-modules/consolekit/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    homepage = "https://pypi.org/project/consolekit";
```
Comment (dotlambda) [inline pkgs/development/python-modules/consolekit/default.nix:38]: GitHub or similar please

**Review by dotlambda — COMMENTED** [review]
Why was this merged? Please check my linked PR for how to do things properly and fix asap

**Comment by dotlambda** [comment]
One commit for many packages??

**Pull request merged**